pyext/src/analysis_trajectories.py

```diff
@@ -42,25 +43,69 @@ class AnalysisTrajectories(object):
-                 nproc=6,
-                 nskip=200):
+                 analysis_dir = 'analysis',
+                 detect_equilibration=True,
+                 burn_in=1000,
+                 nskip=200,
+                 nproc=1,
+                 plot_fmt='pdf'):
+        
+        '''
+        Analyze the ensemble of models obtained after structural sampling by
+        filtering out the bad scoring models, and clustering the good scoring
+        models according to one or multiple score terms, thereby obtaining
+        a set of representative good scoring models that can be used to
+        subsequently compute sampling precision and model precision [using the
+        sampcon module].
+
+        @param out_dirs: list of directories containing the trajectory (rmf
+        files and stat files) for each independent run.
+        
+        @param dir_name: prefix for all directories containing structural 
+        sampling runs. Usually these are named as <prefix>1, <prefix>2, etc.
+        Default is "run_"
+        
+        @analysis_dir: name of output directory where all analysis results will
+        be stored. Default is "analysis".
+        
+        @detect_equilibration: if True, the equilibration period is rigorously
+        determined from the trajectory of all scores using statistical inefficiency measures. Default is True.
+        
+        @burn_in: a total of <burn_in> number of frames are always discarded
+        from the beginning of the trajectory, when calculating statistics.
+        Default is 1000.
+        
+        @nskip: skips every <nskip> frames when calculating statistics.
+        Default is 200.
+        
+        @nproc: number of processors to work on parallely (uses the python multiprocess module). Default is 1.
+        
+        @plot_fmt: file extensions for all plots created. Default is pdf.
+        '''
 
-        self.out_dirs = out_dirs
+        self.out_dirs = [os.path.abspath(d) for d in out_dirs]
         self.dir_name = dir_name
-        self.analysis_dir = analysis_dir
+        self.analysis_dir = os.path.abspath(analysis_dir)
+        self.detect_equilibration = detect_equilibration
         self.nproc = nproc
         self.nskip = nskip
+        self.th = burn_in
+        self.plot_fmt = plot_fmt
+
         self.restraint_names = {}
         self.all_score_fields = []
-
-        self.th = 1000
         self.rerun = False
-
+        
+        # report if equilibration detection has been requested
+        if not self.detect_equilibration:
+            print('Not running rigorous equilibration check.')
+        
+        # report burn-in
+        if self.th > 0:
+            print('Discarding %d (burn-in) frames from the beginning of each independent run' % self.th)
+                
         # Create analysis dir if missing
-        if not os.path.exists(self.analysis_dir):
-            os.makedirs(self.analysis_dir)
+        os.makedirs(self.analysis_dir, exist_ok=True)
```

Comment:
This won't work in Python 2 because `exist_ok` doesn't, er, exist there. This is likely why the code was written this way originally.

Reply:
changed back to the python 2 compatible form that first runs a ```os.path.isdir``` check.